format/FlightSql.proto

```diff
@@ -867,6 +867,45 @@ enum SqlSupportsConvert {
   SQL_CONVERT_VARCHAR = 19;
 }
 
+/*
+ * Represents a request to retrieve information about data type supported ona Flight SQL enabled backend.
+ * Used in the command member of FlightDescriptor for the following RPC calls:
+ *  - GetSchema: return the schema of the query.
+ *  - GetFlightInfo: execute the catalog metadata request.
+ *
+ * The returned schema will be:
+ * <
+ *   type_name: utf8 not null,
+ *   data_type: int not null,
+ *   column_size: int,
+ *   literal_prefix: utf8,
+ *   literal_suffix: utf8,
+ *   create_params: utf8
+ *   nullable: int not null,
+ *   case_sensitive: bool null,
+ *   searchable: int not null,
+ *   unsigned_attribute: bool,
+ *   fixed_prec_scale: bool null,
+ *   auto_increment: bool,
+ *   local_type_name: utf8,
+ *   minimum_scale: int,
+ *   maximum_scale: int,
+ *   sql_data_type: int not null,
+ *   sql_datetime_sub: int,
+ *   num_prec_radix: int,
+ *   interval_precision. int
+ * >
+ * The returned data should be ordered by data_type and then by type_name.
+ */
+message CommandGetTypeInfo {
+  option (experimental) = true;
+
+  /*
+   * Specifies the data type to search for the info.
+   */
+  optional int32 data_type = 1;
```

Comment:
Where do the valid values of data_type come from?

Reply:
This is a value that can be passed through the client. We've added it thinking in the implementation with the ODBC since the call SQLGetTypeInfo receives a parameter to filter for a specific data type. The JDBC call doesn't receive any parameter that's why we made it optional

Reply:
Ah okay, so it's entirely up to the client to pass a value compatible with the given DB? That's a little unfortunate.

Reply:
Oh I see, it's also an output column. Makes sense.